core.go

```diff
@@ -0,0 +1,83 @@
+package bitbox
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/jmbarzee/bitbox/proc"
+)
+
+// Core offers the central functionality of BitBox.
+// Core supports basic process control and interaction.
+type Core struct {
+	m         sync.Mutex
```

Comment:
nitpick: I don't know if this pattern is used everywhere, but we tend to embed a mutex if it is meant to protect all fields within the struct, and only used named mutexes if they're for protecting a subset of fields.

Reply:
This is awesome! I've never noticed that you could embed the mutext to indicate the scope of the lock. I'll do this if we settle on a `sync.RWMutex` instead of a `sycn.Map`.